internal/namespace/canonicalization.go

```diff
@@ -0,0 +1,151 @@
+package namespace
+
+import (
+	"fmt"
+	"hash/fnv"
+
+	"github.com/dalzilio/rudd"
+	"github.com/rs/zerolog/log"
+
+	"github.com/authzed/spicedb/pkg/graph"
+	core "github.com/authzed/spicedb/pkg/proto/core/v1"
+)
+
+// computeCanonicalCacheKeys computes a map from permission name to associated canonicalized
+// cache key for each non-aliased permission in the given type system's namespace.
+func computeCanonicalCacheKeys(typeSystem *NamespaceTypeSystem, aliasMap map[string]string) (map[string]string, error) {
```

Comment:
A good idea for the future, but I wouldn't want to necessarily store it now, since then it would get cached and take up space. A good thing to add if/when we have a normal schema vs optimized